src/mailroom.py

```diff
@@ -0,0 +1,339 @@
+                  'Average', 'Last'))
+    for donor in o.values():
+        print('{:<20} ${:<15,.2f} {:<7} ${:<15,.2f} ${:<15,.2f}'
+              .format(donor['name'], donor['total_donation'],
+                      donor['num_donations'], donor['avg_donation'],
+                      donor['last_donation']))
+    print()
+
+
+def user_report():
+    donor = CURRENT_DONOR
+    os.system('clear')
+    print('Donor info:')
+    print('{:>25}{}'.format('Name:  ', donor['name']))
+    print('{:>25}${:,.2f}'.format('Total Donation:  ', donor['total_donation']))
+    print('{:>25}{}'.format('Number of Donations:  ', donor['num_donations']))
+    print('{:>25}${:,.2f}'.format('Average Donation:  ', donor['avg_donation']))
+    print('{:>25}${:,.2f}'.format('Last Donation:  ', donor['last_donation']))
+    print()
+
+
+def list_of_donors(donor_dict):
+        ''' print the list of donor names '''
+        os.system('clear')
+        print('\nCurrent Donors:')
+        for donor in donor_dict.values():
+            print('\t\t{}'.format(donor['name']))
+        print()
+
+
+def create_donor():
+    global DONORS
+    global CURRENT_DONOR
+    new_donor = {'name': CURRENT_STATE['new_donor_name'],
+                 'total_donation': 0,
+                 'num_donations': 0,
+                 'avg_donation': 0,
+                 'last_donation': 0}
+    DONORS[CURRENT_STATE['new_donor_name']] = new_donor
+    CURRENT_DONOR = DONORS[CURRENT_STATE['new_donor_name']]
+    print('Created New Donor:')
+    user_report()
+
+
+def update_donor(amount):
+    global CURRENT_DONOR
+    print('updating: {}'.format(CURRENT_DONOR))
+    CURRENT_DONOR['last_donation'] = amount
+    CURRENT_DONOR['total_donation'] += amount
+    CURRENT_DONOR['num_donations'] += 1
+    CURRENT_DONOR['avg_donation'] = CURRENT_DONOR['total_donation']\
+        / CURRENT_DONOR['num_donations']
+    return CURRENT_DONOR['last_donation']
+
+
+def print_letter():
+    os.system('clear')
+    print("""
+    ----------------------------------------------------------------------------
+
+    Dear {},
+    Thank you for your recent donation of ${:,.2f}. Without donors like
+    you the Save the Womp Rats Foundation would not exist.
+
+    Sincerly Yours,
+    Jane Doe
+    Save the Womp Rats Foundation
+
+    ----------------------------------------------------------------------------
+    """
+          .format(CURRENT_DONOR['name'],
+                  CURRENT_DONOR['last_donation']))
+
+
+def prompt_for_input(message):
+    '''"Request user input based on current runmode'''
+    return input(message)
+
+
+DONORS = initialize_donor_dict()
+CURRENT_DONOR = {}
+STATE_DICT = initialize_state_dict()
+CURRENT_STATE = STATE_DICT['state_1']
+
+if __name__ == '__main__':
+    os.system('clear')
+
+    print('Welcome the the Donor Message Creation Center\n\n')
+
+    while True:
+        response = prompt_for_input(CURRENT_STATE['prompt_message'])
+        CURRENT_STATE['valid_responses'](response)
```

Comment:
This is really close to right.  I think you might clean up your code quite a bit by returning the next valid state from the "valid_responses" function rather than mutating CURRENT_STATE globally.  But this is the basic shape of what I wanted to to try to get.